extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -134,13 +157,130 @@ export abstract class AzureAuth implements vscode.Disposable {
 		}
 	}
 
-	public async hydrateAccount(token: Token, tokenClaims: TokenClaims): Promise<AzureAccount> {
-		const tenants = await this.getTenants(token.token);
-		const account = this.createAccount(tokenClaims, token.key, tenants);
+	private getHomeTenant(account: AzureAccount): Tenant {
+		// Home is defined by the API
+		// Lets pick the home tenant - and fall back to commonTenant if they don't exist
+		return account.properties.tenants.find(t => t.tenantCategory === 'Home') ?? account.properties.tenants[0] ?? this.commonTenant;
+	}
+
+	public async refreshAccess(account: AzureAccount): Promise<AzureAccount> {
+		// Deprecated account - delete it.
+		if (account.key.accountVersion !== AzureAuth.ACCOUNT_VERSION) {
+			account.delete = true;
+			return account;
+		}
+		try {
+			const tenant = this.getHomeTenant(account);
+			const tokenResult = await this.getAccountSecurityToken(account, tenant.id, azdata.AzureResource.MicrosoftResourceManagement);
+			if (!tokenResult) {
+				account.isStale = true;
+				return account;
+			}
+
+			return await this.hydrateAccount(tokenResult, this.getTokenClaims(tokenResult.token));
+		} catch (ex) {
+			if (ex instanceof AzureAuthError) {
+				void vscode.window.showErrorMessage(ex.message);
+				Logger.error(ex.originalMessageAndException);
```

Comment:
You should include information about where these errors are coming from (something like `Error refreshing access for account ${account.displayInfo.displayName}`)

(this statement can generally be applied to all logging statements and error messages displayed to the user)